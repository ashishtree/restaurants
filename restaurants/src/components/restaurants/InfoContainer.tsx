import * as React from "react";
import { Restaurants } from "../../_types_/Restaurants/Restaurants";
import StarRatingComponent from "react-star-rating-component";

export interface Props {
  restaurantData: Restaurants[];
  doFavorite: (para: number) => void;
  selectedFilter: string;
}

class InfoContainer extends React.Component<Props> {
  getStatusClassName = (status: string) => {
    return status === "order ahead" ? "order-ahead" : status;
  };

  doFavorite = (restaurantId: number) => {
    this.props.doFavorite(restaurantId);
  };

  render() {
    const { restaurantData } = this.props;

    return restaurantData.map((data: Restaurants) => {
      const statusClass = this.getStatusClassName(data.status);
      return (
        <div key={data.id} className="restaurant-listing">
          <div className="logo">
            <div className="like" onClick={() => this.doFavorite(data.id)}>
              <i
                className={data.favorite ? `fa fa-heart` : `far fa-heart`}
                aria-hidden={true}
              />
            </div>
            <div className="logo-img">
              <img src={data.image} />
            </div>
            <div className="ratings">
              <StarRatingComponent
                name="rating"
                editing={false}
                starCount={5}
                value={data.sortingValues.ratingAverage}
                starColor="#ffb400"
                emptyStarColor="#ffb400"
                renderStarIcon={(index, value) => {
                  return (
                    <span>
                      <i
                        className={
                          index <= value ? "fas fa-star" : "far fa-star"
                        }
                      />
                    </span>
                  );
                }}
                renderStarIconHalf={() => {
                  return (
                    <span>
                      <span style={{ position: "absolute" }}>
                        <i className="far fa-star" />
                      </span>
                      <span>
                        <i className="fas fa-star-half i" />
                      </span>
                    </span>
                  );
                }}
              />
            </div>
          </div>
          <div className="details">
            <h2 className="name">
              {data.name}
              <span className={`opening__status ${statusClass}`}>
                {data.status}
              </span>
            </h2>
            <div className="info">
              <div className="other-info">
                <div className="distance">
                  <b>Distance </b>
                  {data.sortingValues.distance}
                </div>
                <div className="min-cost">
                  <b>Min Cost </b>
                  {data.sortingValues.minCost}
                </div>
                <div className="delivery-cost">
                  <b>Delivery Cost </b>
                  {data.sortingValues.deliveryCosts}
                </div>
                <div
                  className={
                    this.props.selectedFilter === "bestMatch"
                      ? `best-match`
                      : `best-match hide`
                  }
                >
                  <b>Best Match </b>
                  {data.sortingValues.bestMatch}
                </div>
                <div
                  className={
                    this.props.selectedFilter === "newest"
                      ? `newest`
                      : `newest hide`
                  }
                >
                  <b>Newest </b>
                  {data.sortingValues.newest}
                </div>
                <div
                  className={
                    this.props.selectedFilter === "popularity"
                      ? `popularity`
                      : `popularity hide`
                  }
                >
                  <b>Popularity </b>
                  {data.sortingValues.popularity}
                </div>
                <div
                  className={
                    this.props.selectedFilter === "averageProductPrice"
                      ? `average-product-price`
                      : `average-product-price hide`
                  }
                >
                  <b>Average Product Price </b>
                  {data.sortingValues.averageProductPrice}
                </div>
                <div
                  className={
                    this.props.selectedFilter === "topRestaurant"
                      ? `top-restaurant`
                      : `top-restaurant hide`
                  }
                >
                  <b>High Score </b>
                  {data.sortingValues.topRestaurant}
                </div>
              </div>
            </div>
          </div>
        </div>
      );
    });
  }
}

export default InfoContainer;
