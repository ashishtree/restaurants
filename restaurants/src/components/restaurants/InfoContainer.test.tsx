import * as React from "react";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import InfoContainer from "./InfoContainer";
import RestaurantMockData from "../../__mockData__/mockRestaurants.json";
import * as ShallowRenderer from "react-test-renderer/shallow";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const props = {
    doFavorite: jest.fn(),
    restaurantData: RestaurantMockData,
    selectedFilter: "shortStatusName",
  };
  const mockStore = configureStore();
  const store = mockStore({});

  const myShallowRenderer = ShallowRenderer.createRenderer();

  const wrapper = myShallowRenderer.render(
    <Provider store={store}>
      <InfoContainer {...props} />
    </Provider>
  );

  return {
    props,
    wrapper,
  };
}

describe("InfoContent Component Render", () => {
  describe("InfoContent Component Render", () => {
    it("should render InfoContent Component", () => {
      const { wrapper } = setup();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
