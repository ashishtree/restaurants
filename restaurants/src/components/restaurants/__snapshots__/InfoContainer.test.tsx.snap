// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoContent Component Render InfoContent Component Render should render InfoContent Component 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <InfoContainer
    doFavorite={[MockFunction]}
    restaurantData={
      Array [
        Object {
          "favorite": false,
          "id": 1,
          "image": "http://localhost:3006/images/restaurant_logo.jpg",
          "name": "Tanoshii Sushi",
          "sortingValues": Object {
            "averageProductPrice": 1536,
            "bestMatch": 0,
            "deliveryCosts": 200,
            "distance": 1190,
            "minCost": 1000,
            "newest": 96,
            "popularity": 17,
            "ratingAverage": 4.5,
            "shortStatusName": "oa",
            "topRestaurant": 20234.5,
          },
          "status": "open",
        },
      ]
    }
    selectedFilter="shortStatusName"
  />
</Context.Provider>
`;
